platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java

```diff
@@ -3527,6 +3565,11 @@ public int hashCode() {
      * @param object the object to be treated as literal
      */
     public ExpressionLiteral(@NonNull Object object) {
+      if (object instanceof String) {
+        object = unwrapStringLiteral((String) object);
+      } else if (object instanceof Number) {
+        object = ((Number) object).floatValue();
```

Comment:
Noting, that in order to make serialization/deserialization and testing of raw expressions easier I'm converting every `Number` literal to a `Float` to keep it consistent since JSON format isn't integer-aware.